### ISSUE TITLE
[Bug] HttpClient with HttpClientHandler cancellation not working

### ISSUE THREAD

**Issue by heebaek** [xamarin/Xamarin.Forms#10214]
### Description
HttpClient SendAsync cancellation is not working if HttpClient is created with HttpClientHandler

### Steps to Reproduce
1. run a test project
2. click HTTPCLIENT button
3. click CANCEL button -> works fine
4. click HTTPCLIENTHANDLER button
5. click CANCEL -> not working.

### Expected Behavior
cancellation is working 

### Actual Behavior
cancellation is working after so many seconds(more than a minute) if with httpclienthandler

### Basic Information
bug exists  both on xamarin.ios and xamarin.android, any xamarin forms version and any build type of httpclient implementation.

this bug appeared after recent update of VS for mac.

### Reproduction Link
[ct.zip](https://github.com/xamarin/Xamarin.Forms/files/4436505/ct.zip)

**Comment by rmarinho** [comment]
This doesn't seem a Xamarin.Forms related issue.  Works fine on Windows, you only see this issue on MacOS?

**Comment by heebaek** [comment]
tested with iphone and android. attached project is xamarin forms app. not macOS or windows problem.

**Comment by rmarinho** [comment]
I tested on VisualStudio on Android. (just changed the url for google.com) and it worked.

**Comment by heebaek** [comment]
this is problem about cancellation. if you change url to google, sendasync complete before you click a cancel button.

**Comment by rmarinho** [comment]
not in my phone, i tried and when i cancel i see cancel exception alert on my phone.

**Comment by heebaek** [comment]
can i see your version details of visual studio about page?
i think this bug related some framework version.

**Comment by dbSoft-SoftwareSolutions** [comment]
Problem seems to be in HttpClient Implementation. I thinl that in new visual studio editions HttpClient Implementation as Managed is does't supported any more. You need to set it as Android. Or programatically in your code to use AndroidClientHandler instead HttpClientHandler. I have the same issue and i am trying to find a working answer. 
Problem seems to be only in visual studio 2019 not in 2017.
Check this link. https://docs.microsoft.com/en-us/xamarin/android/app-fundamentals/http-stack?tabs=windows

**Comment by heebaek** [comment]
Thanks for your good workaround. if i use Xamarin.Android.Net.AndroidClientHandler from source code, cancellation is working. 
but unfortunately, android handler does not support PROPFIND method.
Please let me know if you find another solution.
Thanks a lot.

**Comment by dbSoft-SoftwareSolutions** [comment]
Sorry i don't know what PROPFIND is. Also in my applications, when i was used AndroidClientHandler i was getting some exceptions in random moments, such as Conneciton Reset and Socket is Closed. I didn't got any of this errors when i used HttpClientHandler. Please let me know if you got any of this errors

**Comment by samhouts** [comment]
@marek-safar This looks like a mono issue to me. Does it look familiar?

**Comment by marek-safar** [comment]
Yeah, this looks like something new/recent in Mono but what version are you using?

/cc @steveisok

**Comment by redradist** [comment]
I've have the same issue with Xamarin.Forms testing on Android ...
It seems like cancellation sometimes works sometimes not ...

Is there a fix for it ?

**Comment by dbSoft-SoftwareSolutions** [comment]
> I've have the same issue with Xamarin.Forms testing on Android ...
> It seems like cancellation sometimes works sometimes not ...
> 
> Is there a fix for it ?

What  HttpClient Implementation are you use?

**Comment by cncb-gh** [comment]
I have the same issue with Xamarin in iOS.  Do we know if it is definitely a Mono issue and if it is being worked on?

**Comment by dorincinipa** [comment]
> From 2019 is this [error ](https://github.com/mono/mono/issues/12577) here. Unfortunately, it is still the case that canceling is not possible under Xamarin. Is there a workaround or a fix for this, since you can't write decent RestAPIs this way?

You can create your own [TimeOutHandler.cs](https://gist.github.com/dorincinipa/b7c41459c37c84d4f9f67c8fdbffea9b) as workaround.
Example:

```
HttpClientHandler handler = new HttpClientHandler();
TimeOutHandler timeOutHandler = new TimeOutHandler(handler, TimeSpan.FromSeconds(15));
HttpClient httpClient = new HttpClient(timeOutHandler);
```

**Comment by van8tka** [comment]
Today 2022 and HttpClient with HttpClientHandler cancellation still not working on Android! Any solution?

**Comment by dbSoft-SoftwareSolutions** [comment]
> Today 2022 and HttpClient with HttpClientHandler cancellation still not working on Android! Any solution?

Hello Ivan. We found that the problem is from server side and not from Mobile application. 
We are using: HttpClient implementation  'Android' and Asp Net Core Web Api. And everything works nice.
